### PR TITLE
docs(v2): add documentation about importing markdown

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

The website has [a hidden Markdown page example](https://docusaurus.io/examples/markdownpageexample). It includes [an example of importing markdown](https://docusaurus.io/examples/markdownpageexample#import-mdx-and-md-files), which is not documented anywhere else on the website.

On Discord people often ask about reusing Markdown files, and I have to give them links to this hidden file. It would be great if it can be documented in the actual documentation.

I'm not sure if I'm putting this section in the right place...

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yep

### CODE REVIEW

File: website/docs/guides/markdown-features/markdown-features-react.mdx [hunk 1]
```diff
@@ -69,3 +71,23 @@ You can also import your own components defined in other files or third-party co
...
+
+<BrowserWindow url="http://localhost:3000">
+
+import Intro from './markdown-features-intro.mdx';
+
+<Intro />
+
+</BrowserWindow>
+
+This way, you can reuse contents among multiple pages and avoid duplicating materials.
```
Comment (slorber) [inline website/docs/guides/markdown-features/markdown-features-react.mdx:93]:
```suggestion
<br />

This way, you can reuse contents among multiple pages and avoid duplicating materials.

:::caution

The table-of-contents does not currently contain the imported Markdown headings. This is a technical limitation that we are trying to solve ([issue](https://github.com/facebook/docusaurus/issues/3915)).

::: 
```

**Review by slorber — APPROVED** [review]
Thanks 👍  just added a warning for the TOC problem

**Comment by Josh-Cena** [comment]
> Thanks 👍 just added a warning for the TOC problem

Yep, I realized that as well when I was testing this feature... I'm still stuck with understanding how the MDX loader works, so can't help rn with this

**Comment by slorber** [comment]
Thanks for prettier 😅 

MDX basically converts md to a React component. In the end, Babel/Webpack just see a regular React comp.

The playground might help to understand: https://mdxjs.com/playground/

**Pull request merged**